Toolkit/Public/Get-RscNasSystem.ps1

```diff
@@ -67,10 +94,19 @@ function Get-RscNasSystem {
 
                 $query.Var.Fid = $Id
             }
-            "List" {
+
+            "NameAndList" {
                 $query = New-RscQueryNas -Operation Systems
-                
                 $query.Field.Nodes[0].VendorType = "FETCH"
+                if ($Name -ne $null) {
+                    $InputObj = New-Object -TypeName RubrikSecurityCloud.Types.Filter
+                    $InputObj.Field = "Name"
+                    $InputObj.Texts = @($Name)
+                    $query.Var.Filter = @($InputObj)
+                }
+                if ($First -ne $null) {
```

Comment:
It's rare, if ever, that an operator wants the first N of anything. Typically they are doing reporting or searching for something, so they are going to want all.

Reply:
Remove this code portion dealing with `first`